lib/util/ts_parser.js

```diff
@@ -1069,6 +1089,40 @@ shaka.util.TsParser = class {
     return ('0' + x.toString(16).toUpperCase()).slice(-2);
   }
 
+  /**
+   * @param {number} value
+   * @param {number} reference
+   * @return {number}
+   * @private
+   */
+  handleRollover_(value, reference) {
+    const MAX_TS = 8589934592;
+    const RO_THRESH = 4294967296;
+
+    let direction = 1;
+
+    if (value > reference) {
+      // If the current timestamp value is greater than our reference timestamp
+      // and we detect a timestamp rollover, this means the roll over is
+      // happening in the opposite direction.
+      // Example scenario: Enter a long stream/video just after a rollover
```

Comment:
This example scenario sounds like it would make a great test case.  Can you add a few test cases?

Reply:
I don't think I have samples to do all the scenarios, but I will add an integration test

Reply:
Is it easier with unit testing?

Reply:
With the integration test is enough.